commons/src/build_output.rs

```diff
@@ -0,0 +1,716 @@
+                .and_then(|output| fun_run::nonzero_streamed(name, output));
+
+            println!(); // Weird output from prior stream adds indentation that's unwanted
+
+            let duration = start.elapsed();
+            let time = fmt::details(time::human(&duration));
+            match output {
+                OutputConfig::Stream => {
+                    section.say(format!("Done {time}"));
+                }
+                OutputConfig::StreamNoTiming => section.say("Done {time}"),
+                OutputConfig::Silent | OutputConfig::InlineProgress => unreachable!(),
+            }
+
+            result
+        }
+    }
+
+    /// Specify how you want a command to be run by `Section::run`
+    pub struct RunCommand<'a> {
+        command: &'a mut Command,
+        name: String,
+        output: OutputConfig,
+    }
+
+    impl<'a> RunCommand<'a> {
+        /// Generate a new `RunCommand` with a different name
+        #[must_use]
+        pub fn with_name(self, name: impl AsRef<str>) -> Self {
+            let name = name.as_ref().to_string();
+            let RunCommand {
+                command,
+                name: _,
+                output,
+            } = self;
+
+            Self {
+                command,
+                name,
+                output,
+            }
+        }
+
+        /// Announce and stream the output of a command
+        pub fn stream(command: &'a mut Command) -> Self {
+            let name = fun_run::display(command);
+            Self {
+                command,
+                name,
+                output: OutputConfig::Stream,
+            }
+        }
+
+        /// Announce and stream the output of a command without timing information at the end
+        pub fn stream_without_timing(command: &'a mut Command) -> Self {
+            let name = fun_run::display(command);
+            Self {
+                command,
+                name,
+                output: OutputConfig::StreamNoTiming,
+            }
+        }
+
+        /// Do not announce or stream output of a command
+        pub fn silent(command: &'a mut Command) -> Self {
+            let name = fun_run::display(command);
+            Self {
+                command,
+                name,
+                output: OutputConfig::Silent,
+            }
+        }
+
+        /// Announce a command inline. Do not stream it's output. Emit inline progress timer.
+        pub fn inline_progress(command: &'a mut Command) -> Self {
+            let name = fun_run::display(command);
+            Self {
+                command,
+                name,
+                output: OutputConfig::InlineProgress,
+            }
+        }
+    }
+
+    enum OutputConfig {
+        Stream,
+        StreamNoTiming,
+        Silent,
+        InlineProgress,
+    }
+}
+
+pub mod fmt {
+    use indoc::formatdoc;
+    use std::fmt::Display;
+
+    pub(crate) const RED: &str = "\x1B[31m";
```

Comment:
You may already know this, but this will choose red, without overriding the font weight. It's possible that this would render bold red if something else set the weight to bold. If you want to explicitly make this not-bold, not-italic, not-underlined with a red foreground, you could use `\x1B[0;31m`.